app/src/full/kotlin/tunnel/DnsProxy.kt

```diff
@@ -59,25 +61,37 @@ internal class DnsProxy(
             w("failed reading DNS message", e)
             return
         }
+        dnsMessage.question.name
         if (dnsMessage.question == null) return
 
         val host = dnsMessage.question.name.toString(true).toLowerCase(Locale.ENGLISH)
         if (blockade.allowed(host) || !blockade.denied(host)) {
             val proxiedDns = DatagramPacket(udpRaw, 0, udpRaw.size, destination.getAddress(),
                     destination.getPort())
             forward(proxiedDns, originEnvelope)
-            emit(TunnelEvents.REQUEST, Request(host))
+            RequestLog.add(ExtendedRequest(host))
         } else {
             dnsMessage.header.setFlag(Flags.QR.toInt())
             dnsMessage.header.rcode = Rcode.NOERROR
             generateDnsAnswer(dnsMessage, denyResponse)
             toDevice(dnsMessage.toWire(), -1, originEnvelope)
-            emit(TunnelEvents.REQUEST, Request(host, blocked = true))
+            RequestLog.add(ExtendedRequest(host, blocked = true))
         }
     }
 
     override fun toDevice(response: ByteArray, length: Int, originEnvelope: Packet?) {
         originEnvelope as IpPacket
+
+        if (isAnswerValid(response)){
+            v("easytosearchstring1 valid Answer")
+        }else{
+            val responseName = Name(response.sliceArray(12 until response.size)).canonicalize().toString(true)
```

Comment:
Why 12?

Reply:
Added a comment:
// the first 12 bytes of response contain the DNS-header and are cut of. The header
// should be followed by a resource record which starts with the name that was requested.